### DISCUSSION TITLE
Building the DAW for Android

### DISCUSSION THREAD

**Discussion by WormSnacks** [FundamentalFrequency/.github#42] [General]
Hello! After some conversations on the Discord I've started poking around at the possibility of building the daw as an apk so that people could potentially replace the pi and screen part of the build with their phone which would cut costs quite a bit with the current pi shortage. After exploring some possibilities I think the most straightforward method is going to be building a Native Activity app with VS which I've been slowly poking at over the last day or so. 

That being said, this is all still very new to me, if there's anyone out there with potential insight into this process or experience with building Android apps with C++ I would love some tips or resources!

**Comment by stonepreston** [comment]
JUCE and the tracktion engine support android. So in theory you should be able to build the DAW for android, although I’m not too sure how the MIDI stuff will work. I think that’s going to be the main challenge. Never done much android stuff though so can’t really say. I would say look into building a simple JUCE app for android first.

**Reply by WormSnacks** [reply]
afaik getting the midi messages to the daw shouldn't be too hard if we're just plugging the teensy straight into the phone rather than to a pi.

I know that Projucer had a really streamlined process for building to Android but I kind of wrote that off since there wasn't a .jucer file for the daw (assuming you wrote it in vscode based on the gitignore?) But there might be something to trying to Frankenstein the code they use for the java wrapper into working here now that I think of it

**Reply by stonepreston** [reply]
The DAW is built with CMake which JUCE fully supports
